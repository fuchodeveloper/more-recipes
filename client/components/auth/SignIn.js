import React, { Component } from 'react';
import { render } from 'react-dom';
import Header from '../../components/navigation/Header';

class SignIn extends React.Component {
  render() {
    return (
      <div>
        {/* Header component for navigation */}
        <Header />

        <div className="container margin-top-70">

        <div className="margin-top-bottom-auto">
            <h2 className="text-center">Welcome Back</h2>
        </div>

        <div className="col-md-5 card mx-auto p-4">
            <div className="mx-auto">
                <form action="#" method="post">
                    <div className="row">

                        <div className="col-md-12 mb-2">
                            <label for="emailAddress" className="forms-label-color">Email Address</label>
                            <input type="text" className="form-control" id="emailAddress" name="emailAddress"/>
                        </div>

                        <div className="col-md-12 mb-2">
                            <label for="password" className="forms-label-color">Password</label>
                            <input type="text" className="form-control" name="password" id="password"/>
                        </div>


                        <div className="col-md-12 mb-2">
                            <label for="submit" className="forms-label-color"/>
                            <input type="submit" id="submit" value="Sign In" className="btn btn-primary form-control btn-primary-color"/>
                            <p className="small text-muted mt-3 text-center">Don't have an account? <a href="sign-up.html">Sign Up</a></p>
                        </div>

                    </div>
                </form>
            </div>
        </div>

        <div className="clearfix m-5"/>
        
      </div>

      {/* End div */}
      </div>
    );
  }
}

export default SignIn;
