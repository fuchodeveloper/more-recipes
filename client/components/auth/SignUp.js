import React, { Component } from 'react';
import { render } from 'react-dom';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Header from '../../components/navigation/Header';

class SignUp extends Component {

  constructor(props) {
    super(props);
    // Initialize state properties
    this.state = { 
      credentials: {
        emailAddress: '',
        password: ''
      }
     }
    //  Bin create user function
    this.createUser = this.createUser.bind(this);
  }

  createUser(event) {
    event.preventDefault();
    console.log(event);
  }

  render() {
    return (
      <div>
          {/* Header component for navigation */}
        <Header />

        <div className="container margin-top-70">
        <div className="margin-top-bottom-auto">
            <h2 className="text-center">Create an account</h2>
        </div>
        <div className="col-md-5 card mx-auto p-4 mb-5">
            <div className="mx-auto">
                <form onSubmit={ this.createUser }>
                    <div className="row">
                        <div className="col-md-12 mb-3">
                            <label for="firstName" className="forms-label-color">First Name</label>
                            <input type="text" id="firstName" name="firstName" className="form-control"/>
                        </div>

                        <div className="col-md-12 mb-2">
                            <label for="lastName" className="forms-label-color">Last Name</label>
                            <input type="text" id="lastName" name="lastName" className="form-control"/>
                        </div>

                        <div className="col-md-12 mb-2">
                            <label for="emailAddress" className="forms-label-color">Email Address</label>
                            <input type="text" id="emailAddress" name="emailAddress" className="form-control"/>
                        </div>

                        <div className="col-md-12 mb-2">
                            <label for="password" className="forms-label-color">Password</label>
                            <input type="text" id="password" name="password" className="form-control"/>
                        </div>

                        <div className="col-md-12 mb-2">
                            <label for="confirmPassword" className="forms-label-color">Confirm Password</label>
                            <input type="text" id="confirmPassword" name="confirmPassword" className="form-control"/>
                        </div>

                        <div className="col-md-12 mb-2">
                            <label className="forms-label-color"></label>
                            <input type="submit" value="Sign Up" className="btn btn-primary form-control btn-primary-color"/>
                            <p className="small text-muted mt-3 text-center">Already have an account? <a href="sign-in.html">Sign in</a></p>
                        </div>
                    </div>
                </form>
            </div>
          </div>
        </div>

        <div className="clearfix m-5"></div>

        {/* End div */}
      </div>
    )
  }
}

export default SignUp;

