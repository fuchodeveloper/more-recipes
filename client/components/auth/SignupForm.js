import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import validateInput from '../shared/validations/signup';

class SignupForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      emailAddress: '',
      password: '',
      password_confirmation: '',
      errors: {},
      isLoading: false
    }
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  isValid() {
    const { errors, isValid } = validateInput(this.state);
    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }

  onSubmit(e) {
    e.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {}, isLoading: true });
          
      this.props.userSignupRequest(this.state)
      .then(() => {
        this.props.addFlashMessage({
          type: 'success',
          text: 'Sign up successful!'
        });
        this.context.router.history.push('/');
      })
      .catch((error) => {
        this.setState({ errors: error.response.data, isLoading: false })
      })
    }

  }

  render() {
    const { errors } = this.state;
    return (
      <form onSubmit={ this.onSubmit }>
        <div className="row"> 
             <div className={classnames("col-md-12 mb-3", {"has-error": errors.firstName})}>
                <label htmlFor="firstName" className="forms-label-color">First Name</label>
                <input 
                  value={this.state.firstName} 
                  type="text"
                  id="firstName" 
                  name="firstName" 
                  className="form-control" 
                  onChange={ this.onChange }
                />
                { errors.firstName && <span className="form-text">{ errors.firstName }</span> }
            </div>

            <div className={classnames("col-md-12 mb-3", {"has-error": errors.lastName})}>
                <label htmlFor="firstName" className="forms-label-color">Last Name</label>
                <input 
                  value={this.state.lastName} 
                  type="text"
                  id="lastName" 
                  name="lastName" 
                  className="form-control" 
                  onChange={ this.onChange }
                />
                { errors.lastName && <span className="form-text">{ errors.lastName }</span> }
            </div>

            <div className={classnames("col-md-12 mb-2", { "has-error": errors.emailAddress })}>
                <label htmlFor="emailAddress" className="forms-label-color">Email Address</label>
                <input 
                  value={this.state.emailAddress} 
                  type="email" 
                  id="emailAddress" 
                  name="emailAddress" 
                  className="form-control" 
                  onChange={ this.onChange }
                />
                { errors.emailAddress && <span className="form-text">{ errors.emailAddress }</span> }
            </div>

            <div className={classnames("col-md-12 mb-2", { "has-error": errors.password })}>
                <label htmlFor="password" className="forms-label-color">Password</label>
                <input 
                  value={this.state.password} 
                  type="password" 
                  id="password" 
                  name="password" 
                  className="form-control" 
                  onChange={ this.onChange }
                />
                { errors.password && <span className="form-text">{ errors.password }</span> }
            </div>

            <div className={classnames("col-md-12 mb-2", { "has-error": errors.password_confirmation })}>
                <label htmlFor="password_confirmation" className="forms-label-color">Confirm Password</label>
                <input 
                  value={this.state.password_confirmation} 
                  type="password" 
                  id="confirmPassword" 
                  name="password_confirmation" 
                  className="form-control" 
                  onChange={ this.onChange }
                />
                { errors.password_confirmation && <span className="form-text">{ errors.password_confirmation }</span> }
            </div>

            <div className="col-md-12 mb-2">
                <label className="forms-label-color"></label>
                {/* <input type="submit" value="Sign Up" className="btn btn-primary form-control btn-primary-color"/> */}
                <div className="form-group"><button  className="btn btn-primary btn-lg" disabled={this.state.isLoading }>Login</button></div>
                <p className="small text-muted mt-3 text-center">Already have an account? <a href="sign-in.html">Sign in</a></p>
            </div>
          </div>
      </form>
    )
  }
}

SignupForm.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
  addFlashMessage: PropTypes.func.isRequired
}

SignupForm.contextTypes = {
  router: PropTypes.object.isRequired
}

export default SignupForm;
