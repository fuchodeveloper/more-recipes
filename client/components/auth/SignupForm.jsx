import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Link } from 'react-router-dom';
import validateInput from '../shared/validations/signup';

class SignupForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      emailAddress: '',
      password: '',
      passwordConfirmation: '',
      errors: {},
      isLoading: false
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  isValid() {
    const { errors, isValid } = validateInput(this.state);
    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }

  onSubmit(e) {
    e.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {}, isLoading: true });

      this.props.userSignupRequest(this.state)
        .then(() => {
          this.props.addFlashMessage({
            type: 'success',
            text: 'Sign up successful!'
          });
          this.context.router.history.push('/');
        })
        .catch((error) => {
          this.setState({ errors: error.response.data, isLoading: false });
        });
    }
  }

  render() {
    const { errors } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <div className="row">
          <div className={classnames('col-md-12 mb-3', { 'text-danger': errors.firstName })}>
            <label htmlFor="firstName" className="forms-label-color">First Name</label>
            <input
              value={this.state.firstName}
              type="text"
              id="firstName"
              name="firstName"
              className="form-control"
              onChange={this.onChange}
            />
            { errors.firstName && <span className="form-text">{ errors.firstName }</span> }
          </div>

          <div className={classnames('col-md-12 mb-3', { 'text-danger': errors.lastName })}>
            <label htmlFor="firstName" className="forms-label-color">Last Name</label>
            <input
              value={this.state.lastName}
              type="text"
              id="lastName"
              name="lastName"
              className="form-control"
              onChange={this.onChange}
            />
            { errors.lastName && <span className="form-text">{ errors.lastName }</span> }
          </div>

          <div className={classnames('col-md-12 mb-2', { 'text-danger': errors.emailAddress })}>
            <label htmlFor="emailAddress" className="forms-label-color">Email Address</label>
            <input
              value={this.state.emailAddress}
              type="email"
              id="emailAddress"
              name="emailAddress"
              className="form-control"
              onChange={this.onChange}
            />
            { errors.emailAddress && <span className="form-text">{ errors.emailAddress }</span> }
          </div>

          <div className={classnames('col-md-12 mb-2', { 'text-danger': errors.password })}>
            <label htmlFor="password" className="forms-label-color">Password</label>
            <input
              value={this.state.password}
              type="password"
              id="password"
              name="password"
              className="form-control"
              onChange={this.onChange}
            />
            { errors.password && <span className="form-text">{ errors.password }</span> }
          </div>

          <div className={classnames('col-md-12 mb-2', { 'text-danger': errors.passwordConfirmation })}>
            <label htmlFor="passwordConfirmation" className="forms-label-color">Confirm Password</label>
            <input
              value={this.state.passwordConfirmation}
              type="password"
              id="confirmPassword"
              name="passwordConfirmation"
              className="form-control"
              onChange={this.onChange}
            />
            { errors.passwordConfirmation && <span className="form-text">{ errors.passwordConfirmation }</span> }
          </div>

          <div className="col-md-12 mb-2">
            <label className="forms-label-color" />
            <div className="form-group"><button className="btn btn-primary btn-lg btn-primary-color" disabled={this.state.isLoading}>Login</button></div>
            <p className="small text-muted mt-3 text-center">Already have an account? <Link to="/login">Sign in</Link></p>
          </div>
        </div>
      </form>
    );
  }
}

SignupForm.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
  addFlashMessage: PropTypes.func.isRequired
};

SignupForm.contextTypes = {
  router: PropTypes.object.isRequired
};

export default SignupForm;
