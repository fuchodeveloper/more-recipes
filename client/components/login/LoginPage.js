import React, { Component } from 'react';
import { render } from 'react-dom';
import Header from '../../components/navigation/Header';
import LoginForm from './LoginForm';

class LoginPage extends React.Component {
  render() {
    return (
      <div>
        {/* Header component for navigation */}
        <Header />

        <div className="container margin-top-70">

        <div className="margin-top-bottom-auto">
            <h2 className="text-center">Welcome Back</h2>
        </div>

        <div className="col-md-5 card mx-auto p-4">
            <div className="mx-auto">

              <LoginForm />

            </div>
        </div>

        <div className="clearfix m-5"/>
        
      </div>

      {/* End div */}
      </div>
    );
  }
}

export default LoginPage;
