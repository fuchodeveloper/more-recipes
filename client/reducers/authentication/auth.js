import isEmpty from 'lodash/isEmpty';
import { SET_CURRENT_USER, SET_CURRENT_USER_FAIL } from '../../action/types';

const initialState = {
  isAuthenticated: false,
  user: {}
};

export default (state = initialState, action) => {
  switch (action.type) {
    case SET_CURRENT_USER:
      return {
        isAuthenticated: !isEmpty(action.user),
        user: action.user
      };

    case SET_CURRENT_USER_FAIL:
      return {
        isAuthenticated: false,
        user: {},
        error: action.error
      };

    default:
      return state;
  }
};
