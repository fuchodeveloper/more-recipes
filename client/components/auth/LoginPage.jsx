/* eslint-disable max-len, react/no-unused-state */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import validateInput from '../shared/validations/login';
import login from '../../action/authentication/loginAction';

/**
   * @description Login form class
   *
   * @class LoginForm
   *
   * @extends {React.Component}
   */
class LoginPage extends React.Component {
  /**
   * @description Creates an instance of LoginForm.
   *
   * @param {Object} props
   *
   * @memberof LoginForm
   */
  constructor(props) {
    super(props);
    this.state = {
      emailAddress: '',
      password: '',
      errors: {
        emailAddress: '',
        password: '',
        error: ''
      },
    };

    /**
     * Bind functions to this
     */
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  /**
 * @description Implement on change for form fields
 *
 * @param {Object} event
 *
 * @memberof LoginPage
 *
 * @returns {undefined}
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
 * @description Function to handle form submissions
 *
 * @param {Object} event
 *
 * @memberof LoginPage
 *
 * @returns {void}
 */
  onSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.setState({ errors: {} });
      this.props.login(this.state);
    }
  }

  /**
   * @description Handle form validation
   *
   * @returns {Object} isValid
   *
   * @memberof LoginPage
   */
  isValid() {
    const { errors, isValid } = validateInput(this.state);

    if (!isValid) {
      this.setState({ errors });
    }

    return isValid;
  }

  /**
   * @description Render the JSX template
   *
   * @memberof LoginPage
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    if (this.props.auth) {
      return <Redirect to="/" />;
    }

    return (
      <div>
        <div className="container margin-top-70">

          <div className="margin-top-bottom-auto">
            <h2 className="text-center">Welcome Back</h2>
          </div>

          <div className="col-md-5 card mx-auto p-4">
            <div className="mx-auto">

              <form onSubmit={this.onSubmit}>
                <div className="row">

                  <div className="col-md-12 mb-2">
                    <label htmlFor="emailAddress" className="forms-label-color">
            Email Address
                    </label>
                    <input
                      type="text"
                      className="form-control"
                      id="emailAddress"
                      name="emailAddress"
                      value={this.emailAddress}
                      onChange={this.onChange}
                      required
                    />
                    <span className="mb-1" />
                    { this.state.errors.emailAddress && <div className="text-danger text-center">{this.state.errors.emailAddress}</div> }
                  </div>

                  <div className="col-md-12 mb-2">
                    <label htmlFor="password" className="forms-label-color">Password</label>
                    <input
                      type="password"
                      className="form-control"
                      name="password"
                      id="password"
                      value={this.password}
                      onChange={this.onChange}
                      required
                    />
                    <span className="mb-1" />
                    { this.state.errors.password && <div className="text-danger text-center">{this.state.errors.password}</div> }

                  </div>
                  <div className="col-md-12 mb-2">
                    <label htmlFor="submit" className="forms-label-color" />
                    <input
                      type="submit"
                      id="submit"
                      value="Sign In"
                      className="btn btn-primary form-control btn-primary-color"
                      disabled={this.isFetching}
                    />
                    <p className="small text-muted mt-3 text-center">
                    Don&#8242;t have an account? <Link to="/signup">Sign Up</Link>
                    </p>
                  </div>

                </div>
              </form>

            </div>
          </div>

          <div className="clearfix m-5" />


        </div>

        {/* End div */}
      </div>
    );
  }
}

LoginPage.defaultProps = {
  auth: false
};

LoginPage.propTypes = {
  login: PropTypes.func.isRequired,
  auth: PropTypes.bool
};

LoginPage.contextTypes = {
  router: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  errors: state.auth.error,
  isFetching: state.isFetching,
  auth: state.auth.isAuthenticated
});

const mapDispatchToProps = dispatch => ({
  login: user => dispatch(login(user))
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
