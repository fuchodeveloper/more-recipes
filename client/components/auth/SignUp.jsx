import React, { Component } from 'react';
import { render } from 'react-dom';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import validator from 'validator';
import PropTypes from 'prop-types';
import axios from 'axios';
import Header from '../../components/navigation/Header';
import SignupForm from './SignupForm';
import { userSignupRequest } from '../../action/authentication/signupActions';

class SignUp extends Component {
  render() {
    const { userSignupRequest, addFlashMessage } = this.props;
    return (
      <div>
        {/* Header component for navigation */}
        <Header />

        <div className="container margin-top-70">
          <div className="margin-top-bottom-auto">
            <h2 className="text-center">Create an account!</h2>
          </div>
          <div className="col-md-5 card mx-auto p-4 mb-5">
            <div className="mx-auto">
              <SignupForm userSignupRequest={userSignupRequest} />
            </div>
          </div>
        </div>

        <div className="clearfix m-5" />

        {/* End div */}
      </div>
    );
  }
}

SignUp.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
  addFlashMessage: PropTypes.func.isRequired
};

export default connect(null, { userSignupRequest, addFlashMessage })(SignUp);

