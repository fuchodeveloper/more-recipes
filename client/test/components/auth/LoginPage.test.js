/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import { LoginPage } from '../../../components/auth/LoginPage';
// import mockLocalStorage from '../../__mocks__/mockLocalStorage';

configure({ adapter: new Adapter() });

const userDetails = {
  emailAddress: 'john@gmail.com',
  password: 'password'
};

const setup = () => {
  const props = {
    login: jest.fn(),
    auth: false,
    router: undefined
  };

  return shallow(<LoginPage {...props} />);
};

describe('Login component', () => {
  it('should test that the component rendered successfully', () => {
    const wrapper = setup();
    expect(wrapper.find('.welcome-back').text()).toBe('Welcome Back');
  });

  it('should set email address when email address changes', () => {
    const event = {
      target: {
        name: 'emailAddress',
        value: ''
      }
    };

    const wrapper = setup();
    const emailAddress = wrapper.find('#emailAddress');

    event.target.value = 'john@gmail.com';
    emailAddress.simulate('change', event);

    const expectedEmail = 'john@gmail.com';

    expect(wrapper.instance().state.emailAddress).toBe(expectedEmail);
  });

  it('should set password when password changes', () => {
    const event = {
      target: {
        name: 'password',
        value: ''
      }
    };

    const wrapper = setup();
    const password = wrapper.find('#password');

    event.target.value = 'password';
    password.simulate('change', event);

    const expectedPassword = 'password';

    expect(wrapper.instance().state.password).toBe(expectedPassword);
  });

  it('should login user when correct credentials are supplied', () => {
    const event = {
      preventDefault: jest.fn()
    };

    const wrapper = setup();
    const loginForm = wrapper.find('#loginForm');

    wrapper.setState(userDetails);
    loginForm.simulate('submit', event);
  });
});

