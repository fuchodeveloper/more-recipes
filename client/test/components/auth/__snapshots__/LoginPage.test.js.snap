// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page component Login component snapshot should render with right amount of elements 1`] = `undefined`;
